crates/namada/src/ledger/protocol/mod.rs

```diff
@@ -179,31 +178,52 @@ pub fn dispatch_tx<'a, D, H, CA>(
     vp_wasm_cache: &'a mut VpCache<CA>,
     tx_wasm_cache: &'a mut TxCache<CA>,
     wrapper_args: Option<&mut WrapperArgs>,
-) -> Result<TxResult>
+) -> Result<TxResult<Error>>
 where
     D: 'static + DB + for<'iter> DBIter<'iter> + Sync,
     H: 'static + StorageHasher + Sync,
     CA: 'static + WasmCacheAccess + Sync,
 {
     match tx.header().tx_type {
         // Raw trasaction type is allowed only for governance proposals
-        TxType::Raw => apply_wasm_tx(
-            tx,
-            &tx_index,
-            ShellParams {
-                tx_gas_meter,
-                state,
-                vp_wasm_cache,
-                tx_wasm_cache,
-            },
-        ),
+        TxType::Raw => {
+            // No bundles of governance transactions, just take the first one
+            let cmt = tx.first_commitments().ok_or(Error::MissingInnerTxs)?;
+            let result = apply_wasm_tx(
+                BatchedTxRef { tx: &tx, cmt },
+                &tx_index,
+                ShellParams {
+                    tx_gas_meter,
+                    state,
+                    vp_wasm_cache,
+                    tx_wasm_cache,
+                },
+            )?;
+            Ok(TxResult {
+                gas_used: tx_gas_meter.borrow().get_tx_consumed_gas(),
+                wrapper_changed_keys: Default::default(),
+                batch_results: BatchResults(
+                    [(cmt.get_hash(), Ok(result))].into_iter().collect(),
+                ),
+            })
+        }
         TxType::Protocol(protocol_tx) => {
-            apply_protocol_tx(protocol_tx.tx, tx.data(), state)
+            // No bundles of protocol transactions, only take the first one
```

Comment:
Should we support batched of protocol transactions? I'm not even sure if that can be a thing. In case I believe it would require more rework because of the `Protocol` enum

Reply:
We don't need to worry about this for now.